climada/util/lines_polys_handler.py

```diff
@@ -0,0 +1,1073 @@
+        parent-geometry.
+        If 'SUM', the impact is summed over all points in each geometry.
+        The default is 'SUM'.
+
+    Returns
+    -------
+    Impact
+        Impact object with the impact per geometry (rows of exp.gdf). Contains
+        two additional attributes 'geom_exp' and 'coord_exp', the first one
+        being the origninal line or polygon geometries for which impact was
+        computed.
+
+    See Also
+    --------
+    exp_geom_to_pnt: disaggregate exposures
+
+    """
+
+    # disaggregate exposure
+    exp_pnt = exp_geom_to_pnt(
+        exp=exp, res=res,
+        to_meters=to_meters, disagg_met=disagg_met,
+        disagg_val=disagg_val
+        )
+    exp_pnt.assign_centroids(haz)
+
+    # compute point impact
+    impact_pnt = Impact()
+    impact_pnt.calc(exp_pnt, impf_set, haz, save_mat=True)
+
+    # re-aggregate impact to original exposure geometry
+    impact_agg = impact_pnt_agg(impact_pnt, exp_pnt.gdf, agg_met)
+
+    return impact_agg
+
+
+def impact_pnt_agg(impact_pnt, exp_pnt_gdf, agg_met):
+    """
+    Aggregate the impact per geometry.
+
+    The output Impact object contains an extra attribute 'geom_exp'
+    containing the geometries.
+
+    Parameters
+    ----------
+    impact_pnt : Impact
+        Impact object with impact per exposure point (lines of exp_pnt)
+    exp_pnt_gdf : gpd.GeoDataFrame
+        Geodataframe of an exposures featuring a multi-index. First level indicating
+        membership of original geometries, second level the disaggregated points.
+        The exposure is obtained for instance with the disaggregation method
+        exp_geom_to_pnt().
+    agg_met : AggMethod
+        Aggregation method of the point impacts into impact for respective
+        parent-geometry.
+        If 'SUM', the impact is summed over all points in each geometry.
+        The default is 'SUM'.
+
+    Returns
+    -------
+    impact_agg : Impact
+        Impact object with the impact per original geometry. Original geometry
+        additionally stored in attribute 'geom_exp'; coord_exp contains only
+        representative points (lat/lon) of those geometries.
+
+    See also
+    --------
+        exp_geom_to_pnt: exposures disaggregation method
+    """
+
+    # aggregate impact
+    mat_agg = _aggregate_impact_mat(impact_pnt, exp_pnt_gdf, agg_met)
+
+    # write to impact obj
+    impact_agg = set_imp_mat(impact_pnt, mat_agg)
+
+    # add exposure representation points as coordinates
+    repr_pnts = gpd.GeoSeries(
+        exp_pnt_gdf['geometry_orig'][:,0].apply(
+            lambda x: x.representative_point()))
+    impact_agg.coord_exp = np.array([repr_pnts.y, repr_pnts.x]).transpose()
+
+    # Add original geometries for plotting
+    impact_agg.geom_exp = exp_pnt_gdf.xs(0, level=1)\
+        .set_geometry('geometry_orig')\
+            .geometry.rename('geometry')
+
+    return impact_agg
+
+
+def _aggregate_impact_mat(imp_pnt, gdf_pnt, agg_met):
```

Comment:
```suggestion
def _aggregate_impact_mat(imp_pnt, gdf_pnt):
```

Reply:
I do not understand, why should we remove the aggregation method from the aggregation function? commit 88785411f63103d13e515574aded5bcf18e95cb7 proposes a solution.